image_processor/inc/custom_types.h

```diff
@@ -0,0 +1,8 @@
+#ifndef CUSTOM_TYPES
+#define CUSTOM_TYPES
+#include <vector>
+template<typename T>
+using Matrix = std::vector<std::vector<T>>;

```

Comment:
This data structures is not very efficient. I suggest using a one-dimensional array (`std::vector`) to hold rows and columns.